File: rust/src/daemon_paths.rs
//! Daemon socket and cache paths.

use std::path::{Path, PathBuf};

use crate::file_io::load_project_directory;

/// Return the daemon socket path for a repository.
///
/// # Arguments
/// * `root` - Repository root path.
///
/// # Errors
/// Returns `KanbusError` if the project marker is missing.
pub fn get_daemon_socket_path(root: &Path) -> Result<PathBuf, crate::error::KanbusError> {
    let project_dir = load_project_directory(root)?;
    Ok(project_dir.join(".cache").join("kanbus.sock"))
}

/// Return the index cache path for a repository.
///
/// # Arguments
/// * `root` - Repository root path.
///
/// # Errors
/// Returns `KanbusError` if the project marker is missing.
pub fn get_index_cache_path(root: &Path) -> Result<PathBuf, crate::error::KanbusError> {
    let project_dir = load_project_directory(root)?;
    Ok(project_dir.join(".cache").join("index.json"))
}

File: python/src/kanbus/daemon_paths.py
"""Daemon socket and cache paths."""

from __future__ import annotations

from pathlib import Path

from kanbus.project import load_project_directory, resolve_labeled_projects


def get_daemon_socket_path(root: Path) -> Path:
    """Return the daemon socket path for a repository.

    :param root: Repository root path.
    :type root: Path
    :return: Path to daemon socket file.
    :rtype: Path
    """
    resolve_labeled_projects(root)
    project_dir = load_project_directory(root)
    return project_dir / ".cache" / "kanbus.sock"


def get_index_cache_path(root: Path) -> Path:
    """Return the index cache file path for a repository.

    :param root: Repository root path.
    :type root: Path
    :return: Path to index cache file.
    :rtype: Path
    """
    project_dir = load_project_directory(root)
    return project_dir / ".cache" / "index.json"
